### PR TITLE
Set default status in waitpid when pid = 0.

### PR DESCRIPTION
**Pull request by toots**
Refined PR based on comments at #2222

### CODE REVIEW

**Comment by stedolan** [comment]
This looks good to me. @dra27 ?

File: testsuite/tests/lib-unix/common/wait_nohang.ml [hunk 1]
```diff
@@ -15,3 +15,19 @@ let () =
...
+let () =
+  let fd = Unix.openfile "plip" [O_CREAT; O_WRONLY] 0o666 in
+  let pid =
+    Unix.create_process "sleep" [|"sleep"; "0.01"|] Unix.stdin fd Unix.stderr
+  in
+  Unix.close fd;
+  let pid, status = Unix.waitpid [WNOHANG] pid in
+  assert (pid = 0);
+  assert (status = WEXITED 0);
+  while fst (Unix.waitpid [WNOHANG] pid) = 0 do
```
Comment (xavierleroy) [inline testsuite/tests/lib-unix/common/wait_nohang.ml]: This test looks very fragile to me.  On a busy machine, the `sleep 0.01` could finish before the tester (the present OCaml code) gets to run the `Unix.waitpid`.  Could you please write a test that has no race conditions?  E.g.
* Create a pipe
* Run `cat` with stdin = the pipe and stdout = /dev/null
* Check that `Unix.waitpid [WNOHANG] pid` is pid = 0, status = WEXITED 0.
* Close the pipe, causing `cat` to terminate.
* Check again  `Unix.waitpid [WNOHANG] pid`  in a loop like in the current code.

**Review by xavierleroy — COMMENTED** [review]
The code is obviously good.  The test worries me, see comment below.

**Comment by xavierleroy** [comment]
I pushed on your branch  what I think is a better test, and will now give it a round of precheck CI.

**Comment by xavierleroy** [comment]
The Windows ports are unhappy.  Investigation in progress.

**Comment by xavierleroy** [comment]
After much CI and tweaking of the test, I declare success and will merge this PR.

**Comment by toots** [comment]
Great, thanks for this!

**Pull request merged**